core/src/test/java/nl/nn/adapterframework/http/rest/ApiListenerServletTest.java

```diff
@@ -363,6 +364,30 @@ public void apiListenerThatProducesJSON() throws ServletException, IOException,
 		assertNull(result.getErrorMessage());
 	}
 
+	@Test
+	public void apiListenerThatProducesJSONReturnsNoOutput() throws ServletException, IOException, ListenerException, ConfigurationException {
+		// Arrange
+		String uri="/ApiListenerThatProducesJSONReturnsNoOutput/";
+		new ApiListenerBuilder(uri, Methods.POST, null, MediaTypes.JSON).build();
+		HttpServletRequest request = createRequest(uri, Methods.POST, "{}");
+		session = new HashMap<>();
+		session.put(RESPONSE_CONTENT_KEY, null);
+
+		// Act
+		Response result = service(request);
+
+		// Assert
+		assertAll(
+			() -> assertEquals(200, result.getStatus()),
```

Comment:
I think there should be two possiblilities:
status 200, empty string, could have content type 'text/plain'
status 204, no content, no content type